### PR TITLE
Update pull request document with new features

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Add commentary on WIP title prefixes

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

Add the /hold and /hold cancel commands to PR doc

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

---

xref: https://github.com/kubernetes/test-infra/pull/4206 https://github.com/kubernetes/test-infra/pull/4145
/cc @xiangpengzhao

### CODE REVIEW

**Comment by grodrigues3** [comment]
/cc @fejta (I think he worked with you on the /hold idea)

LGTM.

File: contributors/devel/pull-requests.md [hunk 1]
```diff
@@ -142,6 +143,14 @@ Either the [on call contributor](on-call-rotations.md) will manage the merge que
 
 That's the last step. Your PR is now merged.
 
+## Marking Unfinished Pull Requests
+
+If you want to solicit reviews before the implementation of your pull request is complete, you should mark your pull request as a work in progress to ensure that the merge queue does not pick it up and attempt to merge it. You may prefix your pull request title with `WIP` or `[WIP]` to indicate this.
+
+The GitHub robots will add and remove the `do-not-merge/work-in-progress` label as you edit your title. While the label is present, your pull request will not be considered for merging.
+
+If the pull request is not a work in progress, but you still want to hold the merge queue from merging it, use the `/hold` comment command to do so. When you're ready for the queue to consider your PR for merges again, use the `/hold cancel` comment command.
```
Comment (fejta) [inline contributors/devel/pull-requests.md]: Can we please put this at https://github.com/kubernetes/test-infra/blob/master/commands.md as well?

Also my nit suggestion would be to prioritize this recommendation (as it is more flexibile) over WIP

**Comment by fejta** [comment]
LGTM modulo nits

File: contributors/devel/pull-requests.md (hunk 1 above)
Comment (stevekuznetsov) [reply contributors/devel/pull-requests.md]: The PR adding `/hold` updates that page, but has not merged yet.

**Comment by stevekuznetsov** [comment]
@fejta updated, should not merge until `/hold` PR goes in, though

**Comment by fejta** [comment]
Sure wish I would /lgtm but /hold when it merges :grinning:

File: contributors/devel/pull-requests.md [hunk 2]
```diff
@@ -142,6 +143,16 @@ Either the [on call contributor](on-call-rotations.md) will manage the merge que
...
 That's the last step. Your PR is now merged.
 
+## Marking Unfinished Pull Requests
+
+If you want to solicit reviews before the implementation of your pull request is complete, you should hold your pull request to ensure that the merge queue does not pick it up and attempt to merge it. There are two methods to achieve this:
+
+1. You may use the `/hold` and `/hold cancel` comment commands
+2. You may prefix your pull request title with `WIP` or `[WIP]`
+
+The GitHub robots will add and remove the `do-not-merge/hold` label as you use the comment commands and the `do-not-merge/work-in-progress` label as you edit your title. While either label is present, your pull request will not be considered for merging.
```
Comment (xiangpengzhao) [inline contributors/devel/pull-requests.md]: How to remove a `do-not-merge/work-in-progress` label? Can it be removed by removing the `WIP` prefix in the PR title?

File: contributors/devel/pull-requests.md [hunk 3]
```diff
@@ -142,6 +143,16 @@ Either the [on call contributor](on-call-rotations.md) will manage the merge que
 
 That's the last step. Your PR is now merged.
 
+## Marking Unfinished Pull Requests
+
+If you want to solicit reviews before the implementation of your pull request is complete, you should hold your pull request to ensure that the merge queue does not pick it up and attempt to merge it. There are two methods to achieve this:
+
+1. You may use the `/hold` and `/hold cancel` comment commands
```
Comment (xiangpengzhao) [inline contributors/devel/pull-requests.md]: s/and/or ? I'm not sure.

File: contributors/devel/pull-requests.md [hunk 4]
```diff
@@ -142,6 +143,16 @@ Either the [on call contributor](on-call-rotations.md) will manage the merge que
...
+2. You may prefix your pull request title with `WIP` or `[WIP]`
```
Comment (xiangpengzhao) [inline contributors/devel/pull-requests.md]: Is it case sensitive?

**Comment by stevekuznetsov** [comment]
@xiangpengzhao updated for your comments, rebased

**Comment by xiangpengzhao** [comment]
LGTM. Defer to other folks for another review.

**Comment by stevekuznetsov** [comment]
Once the two PRs xref'ed in the OP are merged, we are good to go :)

**Comment by cblecker** [comment]
@stevekuznetsov they both look merged, but are they actually deployed/live/tested?

**Comment by stevekuznetsov** [comment]
Oh whoops I linked the implementation PRs not the "turn it on" PRs. We want:

https://github.com/kubernetes/test-infra/pull/4322
https://github.com/kubernetes/test-infra/pull/4212

**Comment by xiangpengzhao** [comment]
@stevekuznetsov Both `hold` and `WIP` works in K/K now. So I think it's time to push this forward.

See the test:
hold: https://github.com/kubernetes/kubernetes/pull/52174#issuecomment-328254179
WIP: https://github.com/kubernetes/test-infra/pull/4212

@cblecker please double check this and `/lgtm` the PR :)

**Comment by xiangpengzhao** [comment]
BTW, does this comment https://github.com/kubernetes/test-infra/pull/4322#issuecomment-326675565 mean that the plugin has been deployed?

```
@stevekuznetsov: I updated Prow plugins config for you!
```

**Comment by stevekuznetsov** [comment]
Thanks for grabbing this @xiangpengzhao @cblecker -- I was on PTO for a bit there.

**Pull request merged**